test/rubygems/test_gem_specification.rb

```diff
@@ -2961,6 +2967,25 @@ def test_unresolved_specs_with_versions
     assert_output nil, expected do
       specification.reset
     end
+  ensure
+    reset_orig specification
+  end
+
+  def set_orig(cls)
```

Comment:
I would move this into `lib/rubygems/test_utilities.rb` and have documentation on what this is solving/fixing.

Reply:
If we defined some generic helpers like `redefine_singleton_method(klass, method, implementation, &block)` that redefine the method, yield the block given under the new definition, and then reset the original implementation, I think it would make sense to move it to a support file.

But since these helpers are specific to the methods being redefined here (':unresolved_deps` and `:find_all_by_name`), I think it's more readable to leave the helpers in this file.

Reply:
@MSP-Greg There's still one warning I see, not sure if it's there in your enviroment:

```
/home/deivid/Code/rubygems/test/rubygems/test_gem_commands_build_command.rb:289: warning: assigned but unused variable - trust_dir
```

It's printed at test load time, before the tests start running.

Reply:
@deivid-rodriguez 

Thanks for pointing that out.  It's right there in the Appveyor logs.  I was frustrated by the choppy 'dot' section after `# Running:`, didn't even notice it.   I'll fix.